testsuite/features/step_definitions/common_steps.rb

```diff
@@ -1346,3 +1369,90 @@ def server_secret
+  tmp_dir = "/var/cache/cobbler/buildiso"
+  iso_dir = "/var/cache/cobbler"
+  _out, code = $server.run("cobbler buildiso --tempdir=#{tmp_dir} --iso #{iso_dir}/profile_all.iso --distro=#{distro}")
+  raise 'error in cobbler buildiso, check the Cobbler logs' if code.nonzero?
+  profiles = ["orchid", "flame", "pearl"]
+  isolinux_profiles = []
+  cobbler_profiles = []
+  profiles.each do |profile|
+    # get all profiles from Cobbler
+    result_cobbler, code = $server.run("cobbler profile list | grep -o #{profile}")
+    cobbler_profiles.push(result_cobbler) if code.zero?
+    # get all profiles from isolinux.cfg
+    result_isolinux, code = $server.run("cat #{tmp_dir}/isolinux/isolinux.cfg | grep -o #{profile} | cut -c -6 | head -n 1")
+    unless result_isolinux.empty?
+      isolinux_profiles.push(result_isolinux)
+    end
+  end
+  raise 'error during comparison of Cobbler profiles' unless cobbler_profiles == isolinux_profiles
+end
+
+When(/^I run Cobbler buildiso for distro "([^"]*)" and profile "([^"]*)"$/) do |distro, profile|
+  tmp_dir = "/var/cache/cobbler/buildiso"
+  iso_dir = "/var/cache/cobbler"
+  _out, code = $server.run("cobbler buildiso --tempdir=#{tmp_dir} --iso #{iso_dir}/#{profile}.iso --distro=#{distro} --profile=#{profile}")
+  raise 'error in cobbler buildiso, check the Cobbler logs' if code.nonzero?
+end
+
+When(/^I run Cobbler buildiso for distro "([^"]*)" and profile "([^"]*)" without dns entries$/) do |distro, profile|
+  tmp_dir = "/var/cache/cobbler/buildiso"
+  iso_dir = "/var/cache/cobbler"
+  _out, code = $server.run("cobbler buildiso --tempdir=#{tmp_dir} --iso #{iso_dir}/#{profile}.iso --distro=#{distro} --profile=#{profile} --exclude-dns")
+  raise 'error in cobbler buildiso, check the Cobbler logs' if code.nonzero?
+  # TODO Check for missing nameserver entry in isolinux.cfg
+  result, code = $server.run("cat #{tmp_dir}/isolinux/isolinux.cfg | grep -o nameserver")
+  # we have to fail here if the command suceeds
+  raise 'error in Cobbler buildiso, nameserver parameter found in isolinux.cfg but should not be found' if code.zero?
+end
+
+When(/^I run Cobbler buildiso "([^"]*)" for distro "([^"]*)"$/) do |param, distro|
+  # param can either be standalone or airgapped
+  # workaround to get the contents of the buildiso folder
+  step %(I run Cobbler buildiso for distro "#{distro}" and all profiles)
+  tmp_dir = "/var/cache/cobbler/buildiso"
+  iso_dir = "/var/cache/cobbler"
+  source_dir = "/var/cache/cobbler/source_#{param}"
+  $server.run("mv #{tmp_dir} #{source_dir}")
+  $server.run("mkdir -p #{tmp_dir}")
+  _out, code = $server.run("cobbler buildiso --tempdir=#{tmp_dir} --iso #{iso_dir}/#{param}.iso --distro=#{distro} --#{param} --source=#{source_dir}")
+  raise 'error in cobbler buildiso, check the Cobbler logs' if code.nonzero?
+end
+
+And(/^I check Cobbler buildiso ISO "([^"]*)" with xorriso$/) do |name|
+  tmp_dir = "/var/cache/cobbler"
+  _out, code = $server.run("cat >#{tmp_dir}/test_image <<-EOF
```

Comment:
Looks good like that, but just in case you want to use it for other cases, we can also have stored files in upload_files folder inside our testsuite folder. Then we upload the file to the server at the path we want.